src/sdo/data_loader/goes/cache.py

```diff
@@ -126,37 +130,50 @@ def fetch_goes_metadata(start, end, path):
                 logger.error(
                     f"could not download GOES series for {goes_path}", e)
                 #raise e
+        else:
+            logger.debug(
+                f"skipping downloading data for {str(goes_path)} as it is already there..")
+
+    # convert to parquet format to read with dask
+    # https://docs.dask.org/en/latest/generated/dask.dataframe.DataFrame.repartition.html
+    # https://docs.dask.org/en/stable/dataframe-best-practices.html
+    parquet_dir = cache_dir / 'goes_ts.parquet'
+    if not os.path.exists(parquet_dir):
+        ddf = dd.read_csv(str(cache_dir) + '/**/*.csv')
+        ddf["timestamp"] = dd.to_datetime(ddf['timestamp'], format=ISO_FORMAT)
+        ddf = ddf.set_index('timestamp', drop=True)
+        ddf = ddf.repartition(freq='7d')
+        ddf.to_parquet(parquet_dir, overwrite=True)
```

Comment:
parquet is a powerful columnar file format designed for efficient data storage and retrieval

https://parquet.apache.org/